launcher/src/main/java/org/apache/spark/launcher/SparkClassCommandBuilder.java

```diff
@@ -80,12 +81,18 @@
     }
 
     List<String> cmd = buildJavaCommand(extraClassPath);
+
     for (String key : javaOptsKeys) {
-      addOptionString(cmd, System.getenv(key));
+      String envValue = System.getenv(key);
+      if (!isEmpty(envValue) && envValue.contains("Xmx")) {
```

Comment:
Its not a single cmd string that's formed as its a list of string. We would have to iterated through it anyways. So not making the change for this one.